discovery/seata-discovery-redis/src/main/java/io/seata/discovery/registry/redis/RedisRegistryServiceImpl.java

```diff
@@ -253,6 +254,30 @@ List<InetSocketAddress> lookupByCluster(String clusterName) {
         return new ArrayList<>(CLUSTER_ADDRESS_MAP.getOrDefault(clusterName, Collections.emptySet()));
     }
 
+    /**
+     *
+     * if the serverAddr is unique in the address list and
+     * the callback cluster is current cluster, then enable push-empty protection
+     * Otherwise, remove it.
+     *
+     * @param notifyCluserName notifyCluserName
+     * @param serverAddr serverAddr
+     */
+    private void removeServerAddressByPushEmptyProtection(String notifyCluserName, String serverAddr) {
+
+        Set<InetSocketAddress> socketAddresses = CLUSTER_ADDRESS_MAP.getOrDefault(notifyCluserName, Collections.emptySet());
+        InetSocketAddress inetSocketAddress = NetUtil.toInetSocketAddress(serverAddr);
+        if (socketAddresses.size() == 1 && socketAddresses.contains(inetSocketAddress)) {
+            String txServiceGroupName = ConfigurationFactory.getInstance()
+                    .getConfig(ConfigurationKeys.TX_SERVICE_GROUP);
+            String clusterName = getServiceGroup(txServiceGroupName);
```

Comment:
已经有clustername了,为什么还要通过事务分组再次读取?
I already have clustername, why read it again through transaction grouping?

Reply:
notifyCluserName 是注册中心回调通知的clusterName, 需要和当前正在使用的clusterName做对比.
notifyCluserName is the clusterName of the registration center callback notification, which needs to be compared with the clusterName currently in use.

Reply:
> clusterName

为什么要跟当前的做对比？而不是直接用他回调的去删除缓存中的？ 变化的是回调的那个cluster，如果你不需要改变，为什么不是取消对其的订阅？
Why compare it with the current one? Instead of directly using what he called back to delete the one in the cache? What changes is the cluster that called back. If you don't need to change it, why not unsubscribe to it?